crates/sui-rpc-benchmark/src/bin/pull_grafana_logs.rs

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// This script pulls sampled JSON RPC read requests from Grafana, extracts JSON bodies,
+/// and groups them by RPC "method" for later replay and analysis.
+use reqwest::header::ACCEPT;
+use serde::Deserialize;
+use serde_json::Value;
+use std::collections::HashMap;
+use std::env;
+use std::error::Error;
+use std::fs::File;
+use std::io::{BufWriter, Write};
+use std::process;
+use tracing::{debug, error, info};
+
+// Loki has a limit of 10000 logs per request.
+const MAX_LOGS_PER_REQUEST: u64 = 10000;
+
+/// structs below are to mimic the parsed structure of LokiResponse.
+#[derive(Debug, Deserialize)]
+struct LokiResponse {
+    data: LokiData,
+}
+
+#[derive(Debug, Deserialize)]
+struct LokiData {
+    result: Vec<LokiResult>,
+}
+
+#[derive(Debug, Deserialize)]
+struct LokiResult {
+    values: Vec<(String, String)>,
+}
+
+#[derive(Debug, Deserialize)]
+struct GrafanaLog {
+    message: String,
+}
+
+fn extract_body_from_message(message: &str) -> Option<String> {
```

Comment:
Please add doc comments explaining the format that you are parsing here, and for `fetch_logs`.